File: December 2022/27 December 2022/Python/Container With Most Water.py
def maxArea(A,le):
    l = 0
    r = le-1 
    area = 0
    
    while(l<r):
        area = max(area, min(A[l], A[r]) * (r - l))
        if(A[l]<A[r]):
            l+=1
        else:
            r-=1

    return area 

#{ 
 # Driver Code Starts
#Initial Template for Python 3



for _ in range(0,int(input())):
    
    n = int(input())
    l = list(map(int,input().split()))
    
    print(maxArea(l,n))
    
    
# } Driver Code Ends

File: December 2022/27 December 2022/Java/Container With Most Water.java
//{ Driver Code Starts
//Initial Template for Java


import java.io.*;
import java.util.*;
import java.lang.*;


class GFG {
	public static void main (String[] args) throws IOException{
	    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int t = Integer.parseInt(br.readLine().trim()); //Inputting the testcases
		while(t-->0){
		    int n = Integer.parseInt(br.readLine()); // input size of array
		    int arr[] = new int[n];
		    String inputLine[] = br.readLine().trim().split("\\s+");
		    for(int i=0; i<n; i++){
		        arr[i] = Integer.parseInt(inputLine[i]); // input elements of array
		    }
		    Solve T = new Solve();
            long ans = T.maxArea(arr,n);
            System.out.println(ans);
		}
	}
}

// } Driver Code Ends

class Solve{
    
    long maxArea(int A[], int len){
        int l = 0;
    	int r = len -1;
    	long area = 0;
    
    	while (l < r)
    	{
    		area = Math.max(area, Math.min(A[l], A[r]) * (r - l));
    
    		if (A[l] < A[r])
    			l += 1;
    		else
    			r -= 1;
    	}
    	return area;
    }
    
}